### PR TITLE
feat: Set Theme Variant and Layout defaults in config

### LINKED ISSUE #121: [Feature] Change default theme and layout from config
Hi,

It would be nice to be able to change the default layout (collumns or rows) & default color scheme from the config file.

Every new device you open the page on will resort to the default settings while it might be easier to have a default in config.

### PR DESCRIPTION
**Pull request by taigrr**
## Description

Allows default configuration of theme and layout as described in #121. 

Order of precedence is as follows:
- localstorage (user's preference)
- configuration file 
- browser preference (applies to theme only, as there's no media query for layout)
- light theme

Fixes #121 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] I read & comply with the [contributing guidelines](https://github.com/bastienwirtz/homer/blob/master/CONTRIBUTING.md)
- [x] I have tested my code for new features & regressions on both mobile & desktop devices, using the latest version of major browsers. 
- [x] I have made corresponding changes the documentation (README.md).
- [x] I've check my modifications for any breaking change, especially in the `config.yml` file

### CODE REVIEW

**Comment by bastienwirtz** [comment]
thanks for the PR @taigrr! I'll look into it and test it asap.

File: docs/configuration.md [hunk 1]
```diff
@@ -22,10 +22,12 @@ header: true # Set to false to hide the header
 footer: '<p>Created with <span class="has-text-danger">❤️</span> with <a href="https://bulma.io/">bulma</a>, <a href="https://vuejs.org/">vuejs</a> & <a href="https://fontawesome.com/">font awesome</a> // Fork me on <a href="https://github.com/bastienwirtz/homer"><i class="fab fa-github-alt"></i></a></p>' # set false if you want to hide it.
 
 columns: "3" # "auto" or number (must be a factor of 12: 1, 2, 3, 4, 6, 12)
+vlayout: true # default to the vertical layout
 connectivityCheck: true # whether you want to display a message when the apps are not accessible anymore (VPN disconnected for example)
 
 # Optional theming
 theme: default # 'default' or one of the theme available in 'src/assets/themes'.
+theme_use_dark: false # true or false, useful for overriding browser default in new sessions
```
Comment (bastienwirtz) [inline docs/configuration.md:30]: As the related settings is named `colors`, I see something like that: 
```yaml
default_colors: dark # 'light', 'dark', or 'auto' (default, base on browser preference) 
```
It's also a bit more explicit (I like when the config speaks for itself, with little documentation necessary)

File: src/components/DarkMode.vue [hunk 2]
```diff
@@ -11,23 +11,23 @@
...
   created: function () {
-    this.isDark =
+    let isDark =
       "overrideDark" in localStorage
         ? JSON.parse(localStorage.overrideDark)
-        : matchMedia("(prefers-color-scheme: dark)").matches;
-    this.$emit("updated", this.isDark);
+        : this.isDark === null
+        ? matchMedia("(prefers-color-scheme: dark)").matches
+        : this.isDark;
```
Comment (bastienwirtz) [inline src/components/DarkMode.vue:23]: Chaining 2 ternary operator is not the best for readability, could you replace it with `if / else` statements?

File: src/components/LayoutToggle.vue [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+</template>
+
+<script>
+export default {
+  name: "LayoutToggle",
+  props: {
+    name: String,
+    icon: String,
+    iconAlt: String,
+    vlayout: Boolean,
```
Comment (bastienwirtz) [inline src/components/LayoutToggle.vue:17]: Why not keeping the generic component and add a `default` prop? I think you could achieve the same but I might have missed something. Idealy reusable components are better and I was thinking of using the SettingsToggle component for a couple of other options I might add at some point.

**Review by bastienwirtz — CHANGES_REQUESTED** [review]
Thanks for the great contribution @taigrr! A few notes, but it looks good :+1:

File: src/components/DarkMode.vue (hunk 2 above)
Comment (taigrr) [reply src/components/DarkMode.vue:23]: Sure, I was trying to match the syntax of a similar code block already in the project. Not sure where it was but I'll update mine.

**Comment by finzzz** [comment]
@taigrr Why did you close this? I'm excited to see this feature to be implemented though...

**Comment by taigrr** [comment]
@finzzz I didn't see much good it would get merged and I don't have time to work on this project anymore, but these changes are active on my fork. If you want to use that instead or fork my fork and pr the changes back here, be my guest

**Pull request closed without merging**